ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java

```diff
@@ -13,37 +13,36 @@
 import android.os.Bundle;
 import android.view.KeyEvent;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleObserver;
+import androidx.lifecycle.OnLifecycleEvent;
+
 import com.facebook.infer.annotation.Assertions;
 import com.facebook.react.bridge.Callback;
 import com.facebook.react.devsupport.DoubleTapReloadRecognizer;
 import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
 import com.facebook.react.modules.core.PermissionListener;
 
-import javax.annotation.Nullable;
-
 /**
  * Delegate class for {@link ReactActivity} and {@link ReactFragmentActivity}. You can subclass this
  * to provide custom implementations for e.g. {@link #getReactNativeHost()}, if your Application
  * class doesn't implement {@link ReactApplication}.
  */
-public class ReactActivityDelegate {
+public class ReactActivityDelegate implements LifecycleObserver {
 
-  private final @Nullable Activity mActivity;
+  private final @NonNull ReactActivity mActivity;
   private final @Nullable String mMainComponentName;
 
   private @Nullable ReactRootView mReactRootView;
   private @Nullable DoubleTapReloadRecognizer mDoubleTapReloadRecognizer;
   private @Nullable PermissionListener mPermissionListener;
   private @Nullable Callback mPermissionsCallback;
 
-  @Deprecated
-  public ReactActivityDelegate(Activity activity, @Nullable String mainComponentName) {
-    mActivity = activity;
-    mMainComponentName = mainComponentName;
-  }
```

Comment:
Is it safe to delete this constructor now? It caused a few compile errors in the FB codebase, but I fixed them. How about other people's code?

Reply:
it's been deprecated for a while, also activities must extend ReactActivity and everything will work fine.